library/src/main/java/se/ingenuity/databinding/adapter/TabLayoutBindingAdapter.kt

```diff
@@ -0,0 +1,30 @@
+package se.ingenuity.databinding.adapter
+
+import android.os.Build
+import android.view.ViewTreeObserver
+import androidx.annotation.RequiresApi
+import androidx.databinding.BindingAdapter
+import androidx.viewpager2.widget.ViewPager2
+import com.google.android.material.tabs.TabLayout
+import com.google.android.material.tabs.TabLayoutMediator
+
+object TabLayoutBindingAdapter {
+
+    @JvmStatic
+    @RequiresApi(Build.VERSION_CODES.JELLY_BEAN)
+    @BindingAdapter(value = ["viewPager", "pageNames"], requireAll = false)
+    fun TabLayout.setViewPager(viewPager: ViewPager2, pageNames: List<String>?) {
+        viewPager.viewTreeObserver?.addOnGlobalLayoutListener(object :
```

Comment:
Consider using `core-ktx`'s [doOnAttach](https://developer.android.com/reference/kotlin/androidx/core/view/package-summary#(android.view.View).doOnAttach(kotlin.Function1)) which **attaches** exactly once.

As `Tab` accept `CharSequence` as `text`, you should offer `List<out CharSequence>` rather than `List<String>`.

Reply:
If that's equivalent I can do that. I will test it out 👍 
I just want to move some of the data binding adapters we use a lot(and that have been extensively tested in prod) over to this library. This particular one has been copy pasted to 5+ projects.
Planning to make an extension library as well, for our commonly used extensions.